types/node/timers.d.ts

```diff
@@ -40,18 +40,27 @@ declare module 'timers' {
         }
 
         function setTimeout<TArgs extends any[]>(callback: (...args: TArgs) => void, ms?: number, ...args: TArgs): NodeJS.Timeout;
+        // util.promisify no rest args compability
+        // tslint:disable-next-line void-return
+        function setTimeout(callback: (args: void) => void): NodeJS.Timeout;
         namespace setTimeout {
             const __promisify__: typeof setTimeoutPromise;
         }
         function clearTimeout(timeoutId: NodeJS.Timeout): void;
 
         function setInterval<TArgs extends any[]>(callback: (...args: TArgs) => void, ms?: number, ...args: TArgs): NodeJS.Timer;
+        // util.promisify no rest args compability
+        // tslint:disable-next-line void-return
+        function setInterval(callback: (args: void) => void, ms?: number): NodeJS.Timer;
```

Comment:
Not sure how useful this is for `setInterval` since you can only resolve a promise once.

Reply:
Promisified setInterval returns an async iterator.

Reply:
As far as wrapping it in a `new Promise(...)` goes I'd with setInterval, seems extremely niche.

Reply:
I'm not sure but there may be other circumstances apart from Promises where this comes up. [This comment](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/54258#discussioncomment-962327) may be from a non-Promise situation judging by the function name.

I think it may be best to just merge this now to keep it consistent.